### PR TITLE
Replace list attribute with sets

### PR DESCRIPTION
**Pull request by erademacher**
Lists are ordered. If a user changes the order of a list, Terraform will attempt to update the resource. If there are other resources that depend on the list attributes, they'll be destroyed and recreated. That's a big problem for things like CMEK.

A better alternative is to use a set. It's backwards compatible since it's represented the same way in state, but Terraform ignores the order. Terraform automatically sorts set elements alphabetically, so this required updating some test expecteds.

This also means we no longer need to sort regions or other fields according to the plan.

Using the set plan modifiers required pulling in new framework code, so I also took the opportunity to update the framework from 1.3.3 to 1.3.5 for a few minor bug fixes.

**Commit checklist**
- [x] Changelog
- [x] Doc gen (`make generate`)
- [x] Integration test(s)
- [x] Acceptance test(s)
- [x] Example(s)

### CODE REVIEW

**Review by pawalt — COMMENTED** [review]
Have you tested/is there a good way to test this change with a TFstate that uses the list attributes?

File: internal/provider/log_export_config_resource_test.go [hunk 1]
```diff
@@ -200,7 +200,7 @@ func testLogExportConfigResource(t *testing.T, clusterName string, useMock bool)
 					testLogExportConfigExists(logExportConfigResourceName, clusterResourceName),
 					resource.TestCheckResourceAttr(logExportConfigResourceName, "redact", "false"),
 					resource.TestCheckResourceAttr(logExportConfigResourceName, "groups.#", "2"),
-					resource.TestCheckResourceAttr(logExportConfigResourceName, "groups.0.channels.#", "1"),
+					resource.TestCheckResourceAttr(logExportConfigResourceName, "groups.1.channels.#", "1"),
```
[the remark this answers is not in the document]
Comment (erademacher) [reply internal/provider/log_export_config_resource_test.go:203]: The order of the groups. I'm not 100% sure how ordering of complex types works, but I think TF might just compare the attributes in order. The first attribute here is `LogName` and "devops" comes before "sql".

**Comment by erademacher** [comment]
>Have you tested/is there a good way to test this change with a TFstate that uses the list attributes?

With the test framework? No. But I manually tested a few different moderately complex clusters.

**Review by pawalt — APPROVED** [review]
@erademacher all makes sense. LGTM

**Comment by erademacher** [comment]
TFTR!

**Pull request merged**